docs/source/federation/implementing.md

```diff
@@ -200,4 +200,64 @@ server.listen().then(({ url }) => {
 
 By leveraging the `buildService` function, we're able to customize how requests are sent to our federated services. In this example, we return a custom `RemoteGraphQLDataSource`. The datasource allows us to modify the outgoing request with information from the Apollo Server `context` before it's sent. Here, we're adding the `user-id` header to pass an authenticated user id to downstream services.
 
+In a similar vein, the `didReceiveResponse` hook allows us to inspect a service's `response` in order to modify the `context`. The lifecycle of a request to a federated server involves a number of responses, which may contain headers that need to be passed back to the client. Suppose we want to collect the <a href="https://docs.fastly.com/en/guides/getting-started-with-surrogate-keys" target="_blank" rel="noopener noreferrer">Surrogate-Key</a> headers from every response for caching purposes. We can implement this using the `didReceiveResponse` hook and an `ApolloServerPlugin`:
+
+```javascript{12-23,40-49}
+const { ApolloServer } = require('apollo-server');
+const { ApolloGateway, RemoteGraphQLDataSource } = require('@apollo/gateway');
+
+const gateway = new ApolloGateway({
+  serviceList: [
+    { name: 'products', url: 'http://localhost:4001' },
+    // other services
+  ],
+  buildService({ name, url }) {
+    return new RemoteGraphQLDataSource({
+      url,
+      didReceiveResponse(response, request, context) {
+        if (response.ok) {
+          // Parse surrogate keys from the header and push them onto the context
+          const surrogateKeyString = response.headers.get('Surrogate-Key');
+          if (surrogateKeyString) {
+            context.surrogateKeys.push(...surrogateKeyString.split(' '));
+          }
+          return this.parseBody(response);
+        } else {
+          throw await this.errorFromResponse(response);
```

Comment:
Hmm, this `await` is inside of a non-`async` function.

Reply:
Great catch, and I'm curious why my linter didn't since I _thought_ that I linted this code.
Resolved in 171a101